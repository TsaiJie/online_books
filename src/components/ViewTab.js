/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react'
import PropTypes from 'prop-types'
import Ionicon from 'react-ionicons'
import { LIST_VIEW, CHART_VIEW } from '../utility'

const generateLinkClass = (current, view) => {
  return [current === view ? 'nav-link active' : 'nav-link', current === view ? '#007bff' : 'black']
}

const ViewTab = ({ activeTab, onTabChange }) => (
  <ul className="nav nav-tabs nav-fill my-4">
    <li className="nav-item">
      <a
        className={generateLinkClass(activeTab, LIST_VIEW)[0]}
        href="#"
        onClick={(event) => {
          event.preventDefault()
          onTabChange(LIST_VIEW)
        }}
      >
        <Ionicon
          className="rounded-circle mr-2"
          fontSize="25px"
          color={generateLinkClass(activeTab, LIST_VIEW)[1]}
          icon={'ios-paper'}
        />
        <span style={{color: generateLinkClass(activeTab, LIST_VIEW)[1]}}>列表模式</span>  
      </a>
    </li>
    <li className="nav-item">
      <a
        className={generateLinkClass(activeTab, CHART_VIEW)[0]}
        href="#"
        onClick={(event) => {
          event.preventDefault()
          onTabChange(CHART_VIEW)
        }}
      >
        <Ionicon
          className="rounded-circle mr-2"
          fontSize="25px"
          color={generateLinkClass(activeTab, CHART_VIEW)[1]}
          icon={'ios-pie'}
        />
        <span style={{color: generateLinkClass(activeTab, CHART_VIEW)[1]}}>图表模式</span>  
      </a>
    </li>
  </ul>
)
ViewTab.propTypes = {
  activeTab: PropTypes.string.isRequired,
  onTabChange: PropTypes.func.isRequired,
}
export default ViewTab
