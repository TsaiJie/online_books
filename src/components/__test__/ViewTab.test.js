import React from 'react'
import { shallow } from 'enzyme'
import ViewTab from '../ViewTab'
import Ionicon from 'react-ionicons'
import { LIST_VIEW, CHART_VIEW } from '../../utility'

const props = {
  activeTab: LIST_VIEW,
  onTabChange: jest.fn()
}
let wrapper 
describe('test ViewTab component', () => {
  beforeEach(()=>{
    wrapper = shallow(<ViewTab {...props} />)
  })
  it('should render the component to match snapshot', () => {
    // wrapper.getElement() 修复empty ShallowWrapper {}
    expect(wrapper.getElement()).toMatchSnapshot()
  })
  it('should generate correct className with a', () => {
    const tabItem = wrapper.find('.nav-item').find('a')
    expect(tabItem.first().props().className).toBe('nav-link active')
    expect(tabItem.last().props().className).toBe('nav-link')
  })
  it('should trigger the correct function callbacks', ()=>{
    const lastTabItem = wrapper.find('.nav-item').find('a').last()
    // { preventDefault: () => {} } 默认事件
    lastTabItem.simulate('click', { preventDefault: () => {}})
    expect(props.onTabChange).toHaveBeenCalledWith(CHART_VIEW)
  })
})