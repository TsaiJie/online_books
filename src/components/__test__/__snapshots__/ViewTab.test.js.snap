// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ViewTab component should render the component to match snapshot 1`] = `
<ul
  className="nav nav-tabs nav-fill my-4"
>
  <li
    className="nav-item"
  >
    <a
      className="nav-link active"
      href="#"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle mr-2"
        color="#007bff"
        fontSize="25px"
        icon="ios-paper"
        rotate={false}
        shake={false}
        style={Object {}}
      />
      <span
        style={
          Object {
            "color": "#007bff",
          }
        }
      >
        列表模式
      </span>
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle mr-2"
        color="black"
        fontSize="25px"
        icon="ios-pie"
        rotate={false}
        shake={false}
        style={Object {}}
      />
      <span
        style={
          Object {
            "color": "black",
          }
        }
      >
        图表模式
      </span>
    </a>
  </li>
</ul>
`;
